src/compiler/checker.ts

```diff
@@ -20601,19 +20601,37 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return !!(entry & RelationComparisonResult.Succeeded);
         }
         const targetEnumType = getTypeOfSymbol(targetSymbol);
-        for (const property of getPropertiesOfType(getTypeOfSymbol(sourceSymbol))) {
-            if (property.flags & SymbolFlags.EnumMember) {
-                const targetProperty = getPropertyOfType(targetEnumType, property.escapedName);
+        for (const sourceProperty of getPropertiesOfType(getTypeOfSymbol(sourceSymbol))) {
+            if (sourceProperty.flags & SymbolFlags.EnumMember) {
+                const targetProperty = getPropertyOfType(targetEnumType, sourceProperty.escapedName);
                 if (!targetProperty || !(targetProperty.flags & SymbolFlags.EnumMember)) {
                     if (errorReporter) {
-                        errorReporter(Diagnostics.Property_0_is_missing_in_type_1, symbolName(property), typeToString(getDeclaredTypeOfSymbol(targetSymbol), /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType));
+                        errorReporter(Diagnostics.Property_0_is_missing_in_type_1, symbolName(sourceProperty), typeToString(getDeclaredTypeOfSymbol(targetSymbol), /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType));
                         enumRelation.set(id, RelationComparisonResult.Failed | RelationComparisonResult.Reported);
                     }
                     else {
                         enumRelation.set(id, RelationComparisonResult.Failed);
                     }
                     return false;
                 }
+                const sourceValue = getEnumMemberValue(getDeclarationOfKind(sourceProperty, SyntaxKind.EnumMember)!);
+                const targetValue = getEnumMemberValue(getDeclarationOfKind(targetProperty, SyntaxKind.EnumMember)!);
+                if (sourceValue !== targetValue || sourceValue === undefined || targetValue === undefined) {
+                    if (!errorReporter) {
+                        enumRelation.set(id, RelationComparisonResult.Failed);
+                    }
+                    else if (sourceValue !== targetValue && sourceValue !== undefined && targetValue !== undefined) {
+                        const escapedSource = typeof sourceValue === "string" ? `"${escapeString(sourceValue)}"` : sourceValue;
+                        const escapedTarget = typeof targetValue === "string" ? `"${escapeString(targetValue)}"` : targetValue;
+                        errorReporter(Diagnostics.Each_declaration_of_0_1_differs_in_its_value_where_2_was_expected_but_3_was_given, symbolName(targetSymbol), symbolName(targetProperty), escapedTarget, escapedSource);
+                        enumRelation.set(id, RelationComparisonResult.Failed | RelationComparisonResult.Reported);
+                    }
+                    else {
+                        errorReporter(Diagnostics.Values_of_0_1_are_not_sufficiently_known, symbolName(targetSymbol), symbolName(targetProperty));

```

Comment:
In what situations can an enum member have an undetermined value? One is when the enum declaration is ambient, but can it also happen for computed members?

Reply:
I believe it also comes up if you have an enum member with an unspecified value that follows a dynamically computed value that's explicit.

Reply:
In other words, [something like this](https://www.typescriptlang.org/play?ssl=6&ssc=2&pln=1&pc=1#code/DYUwLgBAzg9gtiAagQ2AVxBAvBAsssACwDoAnZAOwBN4AKASgG4BYAKDZArTggFEIA3mwgQAgtggBGAEwBmCAGpo8JKgwAaYRABCm1gF8gA):

```ts
let someValue = Math.random();

enum E {
  A = 123 + someValue,
  B,
}
```

Reply:
In the design meeting notes, it looks like we decided to not issue the errors for any undetermined values. Is that accurate? If so, it sounds like we should drop the "not sufficiently known" error.